### PR TITLE
make usage of levels more flexible in controls

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

The idea is to allow explicit inheritance of levels. Level can inherit from more than one level. Also a control can belong to more than one level.

- changes to build system
- changes to ANSSI control file to reflect changes
- update of unit tests
- update of documentation

#### Rationale:

Based on this discussion: https://github.com/ComplianceAsCode/content/discussions/6996#discussioncomment-759112

### CODE REVIEW

File: tests/unit/ssg-module/data/controls_dir/abcd-levels.yml [hunk 1]
```diff
@@ -4,26 +4,32 @@ id: abcd-levels
...
+    inherits_from:
+    - low
 
 controls:
   - id: S1
     title: User session timeout
 
   - id: S2
-    level: low
+    leves:
```
Comment (cipherboy) [inline tests/unit/ssg-module/data/controls_dir/abcd-levels.yml:17]: levels?

File: tests/unit/ssg-module/test_controls.py [hunk 2]
```diff
@@ -55,28 +55,28 @@ def test_controls_levels():
...
-    assert c_3.level == "high"
+    assert c_3.levels == ["high"]
 
     c_4a = controls_manager.get_control("abcd-levels", "S4.a")
-    assert c_4a.level == "low"
+    assert c_4a.levels == ["low"]
 
     c_4a = controls_manager.get_control("abcd-levels", "S4.b")
-    assert c_4a.level == "high"
+    assert c_4a.levels == ["high"]
```
[the remark this answers is not in the document]
Comment (vojtapolasek) [reply tests/unit/ssg-module/test_controls.py]: I believe we are already testing for a level which does not inherit - the "low" level.
I added "medium" level to test chained inheritance.

**Comment by vojtapolasek** [comment]
Hello, thank you for the review. I tried to reword commit messages to make them more helpful.
Support for using non-related levels (by relation I mean inheritance) is supported, you can try it:
1. Build rhel8 product and copy resolved anssi high profile somewhere.
2. Comment the part in anssi.yml file which makes the high profile inherit from the enhanced.
3. Add the following to definition of rhel8 anssi profile:
```
    - anssi:all:enhanced
    ```
4. Build rhel8 content.
5. Compare generated anssi high profile with the one created in step 1.
There should be no changes except for possible differences in order of items.

**Comment by jan-cerny** [comment]
@vojtapolasek Looks good to me. I'm only waiting for the CI to finish.

**Pull request merged**